### PR TITLE
Pee Double You Em RadiatorFan

### PR DESCRIPTION
**Pull request by pro-sper**
The fan was updated to use PWM as opposed to GPIO

### CODE REVIEW

File: include/TMS/dev/RadiatorFan.hpp [hunk 1]
```diff
@@ -1,10 +1,15 @@
 #ifndef TMS_INCLUDE_TMS_DEV_RADIATORFAN_HPP
 #define TMS_INCLUDE_TMS_DEV_RADIATORFAN_HPP
 
+#include "EVT/io/PWM.hpp"
```
Comment (mjmagee991) [inline include/TMS/dev/RadiatorFan.hpp]: Use `<>` for your includes instead of `""`

File: src/TMS/dev/RadiatorFan.cpp [hunk 2]
```diff
@@ -2,16 +2,16 @@
 
 namespace TMS {
 
-RadiatorFan::RadiatorFan(IO::GPIO& gpio) : gpio(gpio) {
-    this->powerOn(false);
+RadiatorFan::RadiatorFan(IO::PWM& pwm) : pwm(pwm) {
+    this->pwm.setDutyCycle(100);// setting the duty cycle to 100% to initially start the pump
+    time::wait(3);              // turning on the pump for 3 milliseconds (must do according to data sheet)
```
Comment (mjmagee991) [inline src/TMS/dev/RadiatorFan.cpp]: This is only necessary for the pump, not the fans, so it should be deleted

File: src/TMS/dev/RadiatorFan.cpp [hunk 3]
```diff
@@ -2,16 +2,16 @@
 
 namespace TMS {
 
-RadiatorFan::RadiatorFan(IO::GPIO& gpio) : gpio(gpio) {
-    this->powerOn(false);
+RadiatorFan::RadiatorFan(IO::PWM& pwm) : pwm(pwm) {
+    this->pwm.setDutyCycle(100);// setting the duty cycle to 100% to initially start the pump
```
Comment (mjmagee991) [inline src/TMS/dev/RadiatorFan.cpp]: For the fans, they should initialize to off, so the duty cycle should be 0

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Two minor issues

**Comment by mjmagee991** [comment]
[PWM Fan Drivers](https://trello.com/c/YTzmF9B4/124-pwm-fan-drivers)

File: src/TMS/dev/RadiatorFan.cpp [hunk 4]
```diff
@@ -2,16 +2,15 @@
...
-    } else {
-        gpio.writePin(IO::GPIO::State::LOW);
+void RadiatorFan::setSpeed(uint16_t speed) {
+    if (speed > MAX_SPEED) {
+        speed = MAX_SPEED;
     }
+    pwm.setDutyCycle(speed);
 }
-
 }// namespace TMS
```
Comment (ishan-renjen) [inline src/TMS/dev/RadiatorFan.cpp:17]: add newline

**Review by ishan-renjen — APPROVED** [review]

**Review by ishan-renjen — APPROVED** [review]

**Review by mjmagee991 — APPROVED** [review]
Nice work

**Pull request merged**